### PR TITLE
Update dev domain to `devthedocs.org`

### PR DESCRIPTION
**Pull request by ericholscher**
Refs https://github.com/readthedocs/readthedocs.org/issues/9310

If this looks good, I will update things for `devthedocs.com` dev on corporate.

### CODE REVIEW

File: readthedocs/settings/docker_compose.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def RTD_EXT_THEME_DEV_SERVER_ENABLED(self):
     @property
     def RTD_EXT_THEME_DEV_SERVER(self):
         if self.RTD_EXT_THEME_DEV_SERVER_ENABLED:
-            return "http://assets.community.dev.readthedocs.io:10001"
+            return "http://assets.www.devthedocs.org:10001"
```
Comment (ericholscher) [inline readthedocs/settings/docker_compose.py:66]: Not 100% sure on this one.

**Comment by agjohnson** [comment]
Is there a reason this is using `www.devthedocs.org` instead of `devthedocs.org`? I'd like to see `www` go away completely, more generally, but also `devthedocs.org` matches our production values more closely.

**Review by humitos — APPROVED** [review]
👍🏼 -- I haven't tested the environment, but changes look good to me.

**Comment by ericholscher** [comment]
@agjohnson No particular reason. We can definitely do it at the root domain.

**Comment by agjohnson** [comment]
Yeah, dropping `www` seems better. I'm not opposed to a different domain for the website, but it's at least domain agnostic in local development.

**Review by agjohnson — APPROVED** [review]

**Comment by ericholscher** [comment]
@readthedocs/core Just a heads up, I'm going to merge this. You should now be able to access local dev at http://devthedocs.org/ 🎉 Please let me know if you hit any issues.

**Pull request merged**